ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -325,7 +333,7 @@ func (r *RayClusterReconciler) rayClusterReconcile(ctx context.Context, request
 	if calculateErr != nil {
```

Comment:
I prefer not to set conditions in the reconcile functions. Instead, we should set conditions in `calculateStatus` based on `reconcileErr`.

```go
func calculateStatus(...) {
    if reconcileErr is PodError {
       // set condition
    }

}
```